### PR TITLE
`chain simulator` force change of epoch

### PR DESCRIPTION
**Pull request by miiu96**
## Reasoning behind the pull request
- Added a new method that will trigger a change of epoch 
## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: node/chainSimulator/chainSimulator.go [hunk 1]
```diff
@@ -278,6 +278,22 @@ func (s *simulator) incrementRoundOnAllValidators() {
...
+	for shardID, node := range s.nodes {
+		err := node.ForceChangeOfEpoch()
+		if err != nil {
+			return fmt.Errorf("force change of epoch shardID-%d: error-%w", shardID, err)
+		}
+	}
+
+	epoch := s.nodes[core.MetachainShardId].GetProcessComponents().EpochStartTrigger().Epoch()
+
+	return s.GenerateBlocksUntilEpochIsReached(int32(epoch + 1))
```
Comment (iulianpascalau) [inline node/chainSimulator/chainSimulator.go:294]: :+1:

File: node/chainSimulator/chainSimulator_test.go [hunk 2]
```diff
@@ -155,6 +155,49 @@ func TestChainSimulator_GenerateBlocksAndEpochChangeShouldWork(t *testing.T) {
...
+	require.Nil(t, err)
+
+	err = chainSimulator.ForceChangeOfEpoch()
+	require.Nil(t, err)
+
+	err = chainSimulator.ForceChangeOfEpoch()
+	require.Nil(t, err)
+
+	err = chainSimulator.ForceChangeOfEpoch()
+	require.Nil(t, err)
```
Comment (iulianpascalau) [inline node/chainSimulator/chainSimulator_test.go:198]: maybe add an assert here that the epoch is 5?

**Review by iulianpascalau — APPROVED** [review]

File: node/chainSimulator/chainSimulator_test.go (hunk 2 above)
Comment (miiu96) [reply node/chainSimulator/chainSimulator_test.go:198]: added

**Review by iulianpascalau — DISMISSED** [review]

File: node/chainSimulator/chainSimulator.go [hunk 3]
```diff
@@ -278,6 +278,22 @@ func (s *simulator) incrementRoundOnAllValidators() {
 	}
 }
 
+// ForceChangeOfEpoch will force the change of current epoch
+// This method will call the epoch change trigger and generate block till a new epoch is reached
```
Comment (sstanculeanu) [inline node/chainSimulator/chainSimulator.go:282]:
```suggestion
// This method will call the epoch change trigger and generate blocks till a new epoch is reached
```

**Review by sstanculeanu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**